openapi-cli/src/main/java/io/ballerina/openapi/generators/openapi/OpenApiConverter.java

```diff
@@ -120,4 +128,57 @@ public void generateOAS3DefinitionsAllService(Path servicePath, Path outPath, St
             }
         }
     }
+
+    /**
+     * This method use for checking the duplicate files.
+     *
+     * @param outPath     output path for file generated
+     * @param openApiName given file name
+     * @return file name with duplicate number tag
+     */
+    private static String checkDuplicateFiles(Path outPath, String openApiName, Boolean isJson) {
+        if (outPath != null && Files.exists(outPath)) {
+            final File[] listFiles = new File(String.valueOf(outPath)).listFiles();
+            if (listFiles != null) {
+                openApiName = checkAvailabilityOfGivenName(openApiName, listFiles, isJson);
+            }
+        }
+        return openApiName;
+    }
+
+    private static String checkAvailabilityOfGivenName(String openApiName, File[] listFiles, Boolean isJson) {
+        for (File file : listFiles) {
+            if (System.console() != null && file.getName().equals(openApiName)) {
+                String userInput = System.console().readLine("There is already a/an " + file.getName() +
+                        " in the location. Do you want to override the file? [y/N] ");
+                if (!Objects.equals(userInput.toLowerCase(Locale.ENGLISH), "y")) {
```

Comment:
If the user wants to overwrite the file, shouldn't we just replace the exiting file with the new content?

Reply:
Yeah, if the user needs to overwrite the file. we will replace old content with new content by creating a new file with the same name with new content.